openedx/core/djangoapps/user_authn/cookies.py

```diff
@@ -13,12 +13,70 @@
 from django.dispatch import Signal
 from django.utils.http import cookie_date
 
+from edx_rest_framework_extensions.auth.jwt import cookies as jwt_cookies
 from openedx.core.djangoapps.user_api.accounts.utils import retrieve_last_sitewide_block_completed
 from student.models import CourseEnrollment
 
+from .waffle import JWT_COOKIES_FLAG
+
+
 CREATE_LOGON_COOKIE = Signal(providing_args=['user', 'response'])
 
 
+LOGGED_IN_COOKIE_NAMES = (
+    # Header and payload sections of a JSON Web Token containing user
+    # information and used as an access token.
+    jwt_cookies.jwt_cookie_header_payload_name(),
+
+    # Signature section of a JSON Web Token.
+    jwt_cookies.jwt_cookie_signature_name(),
+
+    # Refresh token, which can be used to get a new JSON Web Token.
+    jwt_cookies.jwt_refresh_cookie_name(),
+
+    # TODO (ARCH-245): Remove the following deprecated cookies.
+    # Set to 'true' if the user is logged in.
+    settings.EDXMKTG_LOGGED_IN_COOKIE_NAME,
+
+    # JSON-encoded dictionary with user information.
+    settings.EDXMKTG_USER_INFO_COOKIE_NAME,
+)
+
+
+def is_logged_in_cookie_set(request):
+    """Check whether the request has logged in cookies set. """
+    if JWT_COOKIES_FLAG.is_enabled():
```

Comment:
What value does this waffle flag provide?

Reply:
It allows us: (1) to decouple deployment of the code from release (enablement) of the code so we can test, monitor, and rollout as we wish, and (2) to immediately disable in case of any unexpected issues. So it is for mostly Use Case 3 in our [Feature Toggle OEP](https://github.com/edx/open-edx-proposals/blob/master/oeps/oep-0017-bp-feature-toggles.rst#ops).  Although, Use Case 1 may also apply as we push forward these PRs incrementally.

I do agree that we should make sure to remove this toggle once we are done with this work.  I added a comment to [JWT_COOKIES_FLAG](https://github.com/edx/edx-platform/blob/68beb6f6659309e22f0f768b5f96501f2edcd254/openedx/core/djangoapps/user_authn/waffle.py#L13) on its expected use cases and a reference to its [removal ticket](https://openedx.atlassian.net/browse/ARCH-247).

Is that what you were getting at?